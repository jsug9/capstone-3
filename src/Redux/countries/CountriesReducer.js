import { fetchCountries } from '../../API/CovidAPI';
import { formattedDate } from '../../Logic/DateFormatter';
import { getMapUrl } from '../../API/MapsAPI';

const GET_COUNTRIES = 'Capstone-3/countries/GET_COUNTRIES';

const initialState = [];
let isLoading = false;

const getCountries = () => async (dispatch) => {
  if (isLoading) return;
  const data = await fetchCountries();
  const countriesList = data.dates[formattedDate].countries;
  const allCountries = data.total;
  const mappedAllCountries = {
    id: allCountries.name,
    name: 'All Countries',
    image: getMapUrl('world'),
    today_confirmed: allCountries.today_confirmed.toLocaleString(),
  };
  const mappedList = Object.keys(countriesList).map((key) => ({
    id: key,
    name: countriesList[key].name,
    image: getMapUrl(countriesList[key].name.toLowerCase()),
    today_confirmed: countriesList[key].today_confirmed.toLocaleString(),
  }));
  mappedList.unshift(mappedAllCountries);
  dispatch({
    type: GET_COUNTRIES,
    payload: mappedList,
  });
  isLoading = true;
};

const countriesReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_COUNTRIES:
      return action.payload;

    default:
      return state;
  }
};

export { getCountries };
export default countriesReducer;
