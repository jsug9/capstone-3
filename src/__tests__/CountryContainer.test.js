import renderer from 'react-test-renderer';
import MockAppComponent from '../__mocks__/App';
import * as Redux from '../Redux/countries/CountriesReducer';
import '@testing-library/jest-dom';
import countriesArray from '../__mocks__/CountriesArray.mock';

jest.spyOn(Redux, 'getCountries')
  .mockImplementation(() => (Redux.dispatchCountries(countriesArray)));

describe('Render App', () => {
  test('Assert snapshot', () => {
    const appSnap = renderer.create(
      <MockAppComponent />,
    ).toJSON();
    expect(appSnap).toMatchSnapshot();
  });
});
