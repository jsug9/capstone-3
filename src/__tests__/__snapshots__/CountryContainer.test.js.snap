// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Assert snapshot 1`] = `
Array [
  <header>
    <nav>
      <a
        aria-current="page"
        className="backArrowDisabled"
        href="/"
        onClick={[Function]}
      >
        <span />
      </a>
      <h1>
        Covid Tracker
      </h1>
      <div
        className="left-nav"
      >
        <span />
        <span />
      </div>
    </nav>
  </header>,
  <div>
    <div
      className="summary"
    >
      <span />
      <div
        className="summary-left"
      >
        <p
          className="allCountries allCountriesSize"
        />
        <p
          className="allCountriesSize"
        >
          <span />
           
          cases
        </p>
      </div>
    </div>
    <div
      className="searchDiv"
    >
      <h2>
        Stats by
         
        Country
      </h2>
      <input
        className="countryInput"
        onChange={[Function]}
        placeholder="Search Country"
        type="text"
        value=""
      />
    </div>
    <div
      className="countriesList"
    />
  </div>,
]
`;
